pyvolcans/pyvolcans_func.py

```diff
@@ -617,6 +645,133 @@ def match_name(volcano_name):
+
+    # plot single- and total-analogy values for all a priori analogues
+    my_apriori_analogues_plot = \
+        all_my_apriori_analogies.plot.bar(x="name",
+                                          y=["ATs","AG","AM","ASz","ASt"],
+                                          stacked=True)
+
+    fig1 = plt.gcf()
+    my_apriori_analogues_plot.set_ylim([0,1])
+    plt.title(f"A priori analogues: {my_volcano_name} ({my_volcano_vnum})",
+              y=1.15, pad=5)
+    plt.xlabel(None)
+    plt.ylabel('Total Analogy')
+    plt.legend(bbox_to_anchor=(0.9, 1.16), ncol=5)
+    plt.tight_layout() # ensuring labels/titles are displayed properly
+
+    if save_figure:
+        fig1.savefig(
+                (f"{my_volcano_name}_apriori_analogues_"
+                 f"{criteria_weights_text}.png"),
+                dpi=600)
+
+    return my_apriori_analogues_plot
+
+
+def plot_bar_better_analogues(my_volcano_name, my_volcano_vnum,
+                              better_analogues, criteria_weights_text,
+                              save_figure=None):
+    """
+    Plots the percentage of 'better analogues' (i.e. higher value of total
+    analogy with the target volcano) for each of the a priori analogues
+    provided by the user. It optionally saves the figure in png format
+    (600 dpi resolution).
+
+    Parameters
+    ----------
+    my_volcano_name : str
+        Name of volcano introduced by the user (i.e. target volcano).
+    my_volcano_vnum: int
+        Volcano number (VNUM) of volcano introduced by the user in the
+        GVP database (www.volcano.si.edu).
+    better_analogues: dict
+        Dictionary containing the volcano name and percentage* of 'better
+        analogues' for all the a priori analogues provided by the user.
+        *Percentage is calculated as (100 - Percentile) and represents the
+        proportion of volcanoes in the GVP database that are classified as
+        'better analogues' (i.e. higher total analogy) by PyVOLCANS (please
+        see Tierz et al., 2019, and documentation of the function:
+        `get_many_analogy_percentiles()` for more details).
+    criteria_weights_text : str
+        Single string indicating the weighting scheme selected by the user.
+    save_figure : bool, optional
+        Keyword argument that indicates whether the generated figures are to
+        be saved in the current working directory, as indicated by the optional
+        flag `--save_figure` chosen by the user when running PyVOLCANS.
+
+    Returns
+    -------
+    my_better_analogues_plot : matplotlib.container.BarContainer object
+        Instance returned mostly for testing purposes on the function.
+    """
+
+    # open figure for the 'better analogues' bar plot
+    fig2 = plt.figure()
+    my_better_analogues_plot = \
+        plt.bar(range(len(better_analogues)), list(better_analogues.values()),
+                align='center', color='blue', width=0.6)
```

Comment:
Hi @mobiuscreek!

Many thanks for your comments. Could this code above help solve [this issue](https://github.com/BritishGeologicalSurvey/pyvolcans/pull/5#discussion_r619695870)?

All the best